tests/optimizers/botorch_base_test.py

```diff
@@ -155,3 +156,28 @@ def test_find_optimum_in_1d_discrete_space(seed):
     assert (
         sum(l == 0 for l in losses) > 5
     ), "After figuring out the best of the three points, it should only propose that."
+
+
+def test_get_numerical_points_from_discrete_space():
+    p0l, p0h = -5, 10
+    p1 = ("small", "medium", "large")
+    p2 = ("woof", "miaow", "moo")
+    discrete_space = ps.ParameterSpace()
+    discrete_space.add(ps.IntegerParameter("integ", (p0l, p0h)))
+    discrete_space.add(ps.OrdinalParameter("ordin", p1))
+    discrete_space.add(ps.CategoricalParameter("categ", p2))
```

Comment:
The variables for the bounds and values/choices seem to be very basically named and below we hard code their ranges and lengths. To improve this situation overall, how about we go for
```python
p_integ = ps.IntegerParameter("integ", (p0l, p0h))
discrete_space.add(p_integ)
p_ordin = ps.OrdinalParameter("ordin", p1)
discrete_space.add(p_ordin)
p_categ = ps.CategoricalParameter("categ", p2)
discrete_space.add(p_categ)
```
which would enable us to below do the more explicit and maintenance friendly
```python
assert points.shape[0] == p_integ.num_values * p_ordin.num_values * p_categ.num_values
```

Reply:
Great stuff -- thanks! Implemented.